analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/EmptyCollectionsShouldNotBeEnumerated.CSharp8.cs

```diff
@@ -153,3 +153,63 @@ public interface IDecoder
 {
     bool Convert(out int i3, out int i4);
 }
+
+// https://github.com/SonarSource/sonar-dotnet/issues/6179
+public class Repro_6179
```

Comment:
Add some basic tests to demonstrate the logic without all the repro scaffolding. Also, add tests for LINQ-`.Count()`.